server/app/services/grid_service_deployer.rb

```diff
@@ -113,7 +113,7 @@ def deploy_service_instance(total_instances, deploy_futures, instance_number, de
 
     deploy_futures << Celluloid::Future.new {
       instance_deployer = GridServiceInstanceDeployer.new(grid_service_instance_deploy)
-      instance_deployer.deploy(deploy_rev)
+      instance_deployer.deploy(deploy_rev.to_s) # XXX: loss of precision
```

Comment:
This goes badly without the `to_s`, deploys get stuck:

```
D, [2017-11-27T17:08:37.900483 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev= >= 2017-11-27T17:08:35.291635Z
D, [2017-11-27T17:08:38.414106 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev= >= 2017-11-27T17:08:35.291635Z

D, [2017-11-27T17:08:38.919276 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev=2017-11-27 17:08:35 UTC >= 2017-11-27T17:08:35.291635Z
D, [2017-11-27T17:08:39.434186 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev=2017-11-27 17:08:35 UTC >= 2017-11-27T17:08:35.291635Z

D, [2017-11-27T17:08:39.941607 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev=2017-11-27 17:08:35 UTC >= 2017-11-27T17:08:35.291635Z
D, [2017-11-27T17:08:40.443884 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev=2017-11-27 17:08:35 UTC >= 2017-11-27T17:08:35.291635Z
```

The stored timestamps get truncated, and the comparison here fails in a similar way.

Reply:
Changing the `GridServiceInstance.deploy_rev/rev` fields to be `type: Time` doesn't fix it, because it looks like something in the RPC layer (?) truncates the timestamps to three decimal places:

```
D, [2017-11-27T17:13:09.700714 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev=2017-11-27T17:12:55.312000Z >= 2017-11-27T17:12:55.312583Z
D, [2017-11-27T17:13:10.206167 #9] DEBUG -- GridServiceInstanceDeployer: poll service_instance.rev=2017-11-27T17:12:55.312000Z >= 2017-11-27T17:12:55.312583Z
```

Reply:
The `to_s` is probably fine, assuming that we do not have any sub-second deploys...